mapie/conformity_scores/utils.py

```diff
@@ -1,26 +1,199 @@
+        )
+    if method == "cumulated_score":
+        warnings.warn(
+            "WARNING: Deprecated method. "
+            + "The method \"cumulated_score\" is outdated. "
+            + "Prefer to use \"aps\" instead to keep "
+            + "the same behavior in the next release.",
+            DeprecationWarning
+        )
+
+
+def check_target(
+    conformity_score: BaseClassificationScore,
+    y: ArrayLike
+) -> None:
+    """
+    Check that if the type of target is binary,
+    (then the method have to be ``"lac"``), or multi-class.
+
+    Parameters
+    ----------
+    conformity_score: BaseClassificationScore
+        Conformity score function.
+
+    y: NDArray of shape (n_samples,)
+        Training labels.
+
+    Raises
+    ------
+    ValueError
+        If type of target is binary and method is not ``"lac"``
+        or ``"score"`` or if type of target is not multi-class.
+    """
+    check_classification_targets(y)
+    if type_of_target(y) == "binary" and not isinstance(conformity_score, LAC):
+        raise ValueError(
+            "Invalid method for binary target. "
+            "Your target is not of type multiclass and "
+            "allowed values for binary type are "
+            f"{['score', 'lac']}."
+        )
+
+
+def check_classification_conformity_score(
+    conformity_score: Optional[BaseClassificationScore] = None,
+    method: Optional[str] = None,
+) -> BaseClassificationScore:
+    """
+    Check parameter ``conformity_score`` for classification task.
+    By default, return a LAC instance.
+
+    Parameters
+    ----------
+    conformity_score: BaseClassificationScore
+        Conformity score function.
+
+        By default, `None`.
+
+    method: str
+        Method to compute the conformity score.
+
+        By default, `None`.
+
+    Raises
+    ------
+    ValueError
+        If parameters are not valid.
+
+    Examples
+    --------
+    >>> from mapie.conformity_scores.utils import (
+    ...     check_classification_conformity_score
+    ... )
+    >>> try:
+    ...     check_classification_conformity_score(1)
+    ... except Exception as exception:
+    ...     print(exception)
+    ...
+    Invalid conformity_score argument.
+    Must be None or a BaseClassificationScore instance.
+    """
+    if conformity_score is not None:
+        if method is not None:
+            warnings.warn(
+                "WARNING: the `conformity_score` parameter takes precedence "
+                "over the `method` parameter to define the method used.",
+                UserWarning
+            )
+        if isinstance(conformity_score, BaseClassificationScore):
+            return conformity_score
+        else:
+            raise ValueError(
+                "Invalid conformity_score argument.\n"
+                "Must be None or a BaseClassificationScore instance."
+            )
+    elif method is not None:
+        if isinstance(method, str) and method in METHOD_SCORE_MAP:
```

Comment:
no need to check that it's a string as if it's in `METHOD_SCORE_MAP` then it's a string

Reply:
Done 👍

Reply:
But doens't work if the value you test is a list (unshashable value)